paper-tabs.d.ts

```diff
@@ -0,0 +1,208 @@
+   */
+  noBar: boolean|null|undefined;
+
+  /**
+   * If true, the slide effect for the bottom bar is disabled.
+   */
+  noSlide: boolean|null|undefined;
+
+  /**
+   * If true, tabs are scrollable and the tab width is based on the label width.
+   */
+  scrollable: boolean|null|undefined;
+
+  /**
+   * If true, tabs expand to fit their container. This currently only applies when
+   * scrollable is true.
+   */
+  fitContainer: boolean|null|undefined;
+
+  /**
+   * If true, dragging on the tabs to scroll is disabled.
+   */
+  disableDrag: boolean|null|undefined;
+
+  /**
+   * If true, scroll buttons (left/right arrow) will be hidden for scrollable tabs.
+   */
+  hideScrollButtons: boolean|null|undefined;
+
+  /**
+   * If true, the tabs are aligned to bottom (the selection bar appears at the top).
+   */
+  alignBottom: boolean|null|undefined;
+
+  /**
+   * If true, tabs are automatically selected when focused using the
+   * keyboard.
+   */
+  autoselect: boolean|null|undefined;
+
+  /**
+   * The delay (in milliseconds) between when the user stops interacting
+   * with the tabs through the keyboard and when the focused item is
+   * automatically selected (if `autoselect` is true).
+   */
+  autoselectDelay: number|null|undefined;
+  _step: number|null|undefined;
+  _holdDelay: number|null|undefined;
+  _leftHidden: boolean|null|undefined;
+  _rightHidden: boolean|null|undefined;
+  _previousTab: object|null|undefined;
+  readonly _tabContainerScrollSize: any;
+  created(): void;
+  detached(): void;
+  _activateHandler(): void;
+  ready(): void;
+  _noinkChanged(noink: any): void;
+  _setNoinkAttribute(element: any): void;
+  _removeNoinkAttribute(element: any): void;
+  _computeScrollButtonClass(hideThisButton: any, scrollable: any, hideScrollButtons: any): any;
+  _computeTabsContentClass(scrollable: any, fitContainer: any): any;
+  _computeSelectionBarClass(noBar: any, alignBottom: any): any;
+  _onTabSizingChanged(): void;
+  _onIronSelect(event: any): void;
+  _onIronDeselect(event: any): void;
+
+  /**
+   * Activates an item after a delay (in milliseconds).
+   */
+  _scheduleActivation(item: any, delay: any): void;
+
+  /**
+   * Activates the last item given to `_scheduleActivation`.
+   */
+  _delayedActivationHandler(): void;
+
+  /**
+   * Cancels a previously scheduled item activation made with
+   * `_scheduleActivation`.
+   */
+  _cancelPendingActivation(): void;
+  _onArrowKeyup(event: any): void;
+  _onBlurCapture(event: any): void;
+  _scroll(e: any, detail: any): void;
+  _down(e: any): void;
+  _affectScroll(dx: any): void;
+  _onLeftScrollButtonDown(): void;
+  _onRightScrollButtonDown(): void;
+  _onScrollButtonUp(): void;
+  _scrollToLeft(): void;
+  _scrollToRight(): void;
+  _tabChanged(tab: any, old: any): void;
+  _scrollToSelectedIfNeeded(tabWidth: any, tabOffsetLeft: any): void;
+  _calcPercent(w: any, w0: any): any;
+  _positionBar(width: any, left: any): void;
+  _onBarTransitionEnd(e: any): void;
+}
```

Comment:
Is it ok that this type doesn't include the [properties created in `created`](https://github.com/PolymerElements/paper-tabs/blob/48d5a710b6bb386a51b66237e56c37853090d91a/paper-tabs.html#L373-L376)?

Reply:
Hmm, yeah Analyzer doesn't pick up these types of post-construction properties. In this case, they all have leading underscores, which we call "protected" for elements, but it's kinda fuzzy what that means. Since I'm leaning slightly more towards excluding these "protected" properties in the declarations for elements (but not for enough to actually implement that across the board yet), I think we should just ignore these particular ones.